### ISSUE TITLE
Load external content embed by oEmbed with Camo

### ISSUE THREAD

**Issue by Flaburgan** [diaspora/diaspora#6377]
Privacy ftw ;)
##

**Comment by jhass** [comment]
Can you clarify? Camo's upstream commitment is to be an image proxy, not an arbitrary proxy. So even if we can make proxying  arbitrary content work, that may break at any point.

**Comment by Flaburgan** [comment]
We can start with images, that's already an improvement. For example, youtube videos only load a preview from https://i.ytimg.com/ first and then the real content once the user clicked on it. We can assume that if the user wants to see the video, he accepts to be tracked, but at least the users who only scroll their stream won't be.

**Comment by denschub** [comment]
I have checked the terms of service of Twitter and Youtube, and neither do not allow us to alter an embed in any shape. To proxy all requests, we'd somehow have to serve an altered form of the embedding JS (which would be a violation of their copyright, as well as the TOS), and proxy all requests made by this script, which not only would require us to build our own proxy, but also is a violation of their TOS.

Our desire for privacy is neither reason nor permission to violate other service's Terms of Service. The way to go here is to make embeds require user interaction instead, for which I have filed #8142.
